### ISSUE TITLE
Wrap single checkbox for boolean fields in fieldset

### ISSUE THREAD

**Issue by thomasmassmann** [it-spirit/spirit.plone.forms#4]
For better accessibility `checkbox` fields should be wrapped in a `fieldset` with a `legend`. The current implementation (https://github.com/plone/plone.app.z3cform/blob/master/plone/app/z3cform/templates/singlecheckboxbool_input.pt) wraps the checkbox in a `div`

Current implementation for the `plone.versioning` behavior (Plone 6, german):

<img width="733" alt="Screenshot 2021-12-18 at 18 28 15" src="https://user-images.githubusercontent.com/1116199/146658156-2dd034ae-4cf4-49d9-956e-6b42467bd531.png">

The label for the field is the checkbox label (so the answer). This can be confusing for people, as the logical form pattern (label -> field) is broken.

An example is shown in Adam Silver’s book “Form Design Patterns”:

![Single Checkbox for Boolean Fields](https://user-images.githubusercontent.com/1116199/146657995-e02ec5c5-6a0f-493c-924e-e643e580424e.png)

He uses a “question” -> “answer” pattern. This requires an additional setting for the widget in Plone, since the field only has one title and description. The `title` should be used for the `label`, and an additional checkbox title can be defined at the widget. If this one is not available, the `title` would be used a second time. An automatically created text for the answer as fallback if the checkbox title is not available is **not recommended**, as this can lead to unexpected wordings and confusing labels.

Example for the `plone.versioning` behavior (Plone 6, german) with a custom checkbox title:

<img width="714" alt="Screenshot 2021-12-18 at 18 22 57" src="https://user-images.githubusercontent.com/1116199/146658065-4298a2b0-53bc-4bdb-af92-3dccc5fd4398.png">
